code-style/pom.xml

```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+    Copyright (c) 2018 Red Hat Inc and others.
+
+    All rights reserved. This program and the accompanying materials 
+    are made available under the terms of the Eclipse Public License v1.0
+    which accompanies this distribution, and is available at
+    http://www.eclipse.org/legal/epl-v10.html
+    
+    Contributors:
+        Red Hat Inc - initial API and implementation and initial documentation
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>org.eclipse.hono</groupId>
+    <artifactId>hono-parent</artifactId>
+    <version>0.5-M11-SNAPSHOT</version>
+  </parent>
+
+  <artifactId>hono-code-style</artifactId>
```

Comment:
I do not think that we need another module for this purpose. We can add the checkstyle configuration to the already existing `legal` module. Later we might want to rename it to `build-tools` or whatever to better reflect its broadened purpose.

Reply:
So technically we do need a JAR module which is not dependent on `hono-bom` to host the checkstyle configuration. So adding this to `hono-legal` would only be possible if we change the parent to `hono-parent`. As I tried to explain above, this would create a circular dependency otherwise.

Reply:
ok, then we simply define the checkstyle plugin in the parent POM's pluginManagement section and explicitly enable the plugin in the modules that need it. We already do something similar with unpacking the legal docs to the modules.

Reply:
Technically that should work. I am not sure that makes the whole setup simpler. What speaks against having a common parent?

Reply:
you mean, merging the `bom` POM into the parent POM?